probert/tests/test_network.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2025 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+from unittest.mock import Mock, patch
+
+from probert.network import UdevObserver
+
+
+class TestUdevObserver(unittest.TestCase):
+    def test_init_no_nl80211(self):
+        def fake_import(name, globals=None, locals=None, fromlist=(), level=0):
+            if name == "probert" and "_nl80211" in fromlist:
+                raise ImportError
```

Comment:
Unless I misunderstand this test, the goal here is that we might not actually have the _nl80211 module at all, and to fake that case. With current probert that case shouldn't happen, but maybe in the future we don't even want it in the bootstrap snap.  Is that the goal of this test?

Reply:
Actually, I wasn't necessarily thinking about Subiquity / bootstrap when I wrote this test. 

I wanted to start somewhere since we didn't have any test for probert/network.py. And since I moved an import statement, I wanted to make sure I didn't break the existing behavior.

But yes, we could add the option to not have the _nl80211 extension in the snap at all for desktop.